### PR TITLE
Improvement for issue 2167. 

### PR DESCRIPTION
**Pull request by ascoppa**
## Workaround: 

Avoids explicitly rounding dimensions and calling methods that will round dimensions. 

## Note 

See https://github.com/kenwheeler/slick/issues/2167

### CODE REVIEW

File: app/assets/javascripts/jquery.slick.js [hunk 1]
```diff
@@ -308,9 +358,9 @@
         });
 
       } else {
-
         _.applyTransition();
-        targetLeft = Math.ceil(targetLeft);
+
+        targetLeft = Math.floor(targetLeft);
```
Comment (ascoppa) [inline app/assets/javascripts/jquery.slick.js:363]: A couple of notes regarding this call: 

- The need to swap from `ceil` to `floor` was discovered empirically. Using `floor` just does a better job getting rid of the edges on all the scenarios I tested (combined with the [new float implementations](https://github.com/bespokepost/jquery-slick-rails/pull/1/files#diff-b7d0af347012317a0b50d7fa453f22dbbf7a669d684f63957fc428ada1cebfacR191-R238) of width and height, outerHeight and outerWidth).

- I'm sure this is not the case but given that `TargetLeft` is always 0 or a negative value I wonder if the author of this function overlooked that when dealing with negative values `ceil` and `floor` roles are exchanged (if we forget to pay attention to the sign).

Screen Shot 2021-01-04 at 19 59 27

File: app/assets/javascripts/jquery.slick.js [hunk 2]
```diff
@@ -1992,16 +2041,15 @@
...
         positionProps = {},
         x, y;
 
     if (_.options.rtl === true) {
       position = -position;
     }
-    x = _.positionProp == 'left' ? Math.ceil(position) + 'px' : '0px';
-    y = _.positionProp == 'top' ? Math.ceil(position) + 'px' : '0px';
+    x = _.positionProp == 'left' ? Math.floor(position) + 'px' : '0px';
+    y = _.positionProp == 'top' ?  Math.floor(position) + 'px' : '0px';
```
Comment (ascoppa) [inline app/assets/javascripts/jquery.slick.js:2052]: See https://github.com/bespokepost/jquery-slick-rails/pull/1/files#r551619725

File: app/assets/javascripts/jquery.slick.js [hunk 3]
```diff
@@ -2017,46 +2065,45 @@
...
       }
     }
-
   };
 
   Slick.prototype.setDimensions = function() {
-
     var _ = this;
+    var $firstSlide = _.$slides.first();
+    var MAGIC_CONSTANT = 5000;
```
Comment (jackchoi) [inline app/assets/javascripts/jquery.slick.js]: Can use a single `var` declaration to make it more consistent with the style of the [existing code](https://github.com/bespokepost/jquery-slick-rails/blob/546748ea7df856268e28a1500bd6e8633e1aa160/app/assets/javascripts/jquery.slick.js#L735-L737).

**Review by jackchoi — APPROVED** [review]

File: app/assets/javascripts/jquery.slick.js (hunk 3 above)
Comment (ascoppa) [reply app/assets/javascripts/jquery.slick.js]: Jack, addressed here https://github.com/bespokepost/jquery-slick-rails/pull/1/commits/9e7ddf82a0e7d9fc9b17f9e92ca54881d5029e74

**Pull request merged**